src/umpire/strategy/DynamicPool.cpp

```diff
@@ -360,15 +378,14 @@ void DynamicPool::mergeFreeBlocks()
     std::tie(bytes, is_head, whole_bytes) = rec.second;
     insertFree(ptr, bytes, is_head, whole_bytes);
   }
+
+  UMPIRE_LOG(Debug, "() Free blocks after: " << getFreeBlocks());
 }
 
 std::size_t DynamicPool::releaseFreeBlocks()
 {
   UMPIRE_LOG(Debug, "()");
 
-  // Coalesce first so that we are able to release the most memory possible
-  mergeFreeBlocks();
-
```

Comment:
Why is this being removed from here?

Reply:
`releaseFreeBlocks()` is a private method that is only called from `coalesce()` and `allocateFromStrategy()`. `coalesce()` calls `mergeFreeBlocks()` explicitly, so this is an unnecessary operation in that call path.

I agree that this changes the behavior from `allocateFromStrategy()` in the case where it cannot allocate from the resource/strategy. Would you like me to add a call to `mergeFreeBlocks()` there? I'm not sure how much we've tested that ability yet, beyond the relatively simple unit test.

Reply:
Added it!